framework/src/util/Utils.h

```diff
@@ -81,6 +81,16 @@ namespace cppmicroservices
 
     void TerminateForDebug(const std::exception_ptr ex);
 
+    template <typename T>
+    std::shared_ptr<T> atomic_load_tSafe(std::shared_ptr<T> ptr)
+    {
+#ifdef US_ENABLE_THREADING_SUPPORT
+        return std::atomic_load(&ptr);
+#else
+        return ptr;
+#endif
+    }
```

Comment:
this is already implemented by `cppmicroservices::detail::Atomic<T>` in `Threads.h`. I don't think you need another implementation.